content-src/components/DiscoveryStreamComponents/DSCard/DSCard.jsx

```diff
@@ -155,6 +155,9 @@ export class DSCard extends React.PureComponent {
     if (this.observer && this.placholderElement) {
       this.observer.unobserve(this.placholderElement);
     }
+    if (this.idleCallbackId) {
```

Comment:
Won't this always be true? Why are you immediately cancelling the idle callback?

Reply:
Also, why does the idle callback need to be removed if it's only called once?

Reply:
For removing the callback, a component can be mounted, and then unmounted, before the browser does an idle.

For the cases where the component gets unmounted before idle happens, it calls `onIdleCallback` after the component has been unmounted. At best this throws at `setState` at worst, this kinda smells like a memory leak. 

I was initially able to reproduce it but couldn't the next day, but I'm pretty it's there, just hard to reproduce.

For `idleCallbackId` always existing, yeah, it should. I think I'm just being careful here to cover potential refactors that may clear the id, or to cover something I may not be able to predict. Now looking at it, I kinda want to manually clear the id myself inside `onIdleCallback`

Reply:
What are your thoughts on for `idleCallbackId` always existing?

I'm happy to:
1. Remove the if check, like you initially suggested.
2. Leave it a is, after me explaining I'm just being careful and not actually trying to check something.
3. Clear the id so the if in `componentWillUnmount` can actually be avoided if we don't need to cancel it.

Reply:
I actually wasn't looking at this right...I missed that this is part of the unmount method, so I actually think this is probably fine as is. Derp.

Reply:
Once `onIdleCallback` is called I don't think you have to worry about cancelling the idle callback, because AFAIK they only get fired once per ID.

Reply:
Yeah, I did ensure that calling cancel idle on an id that has been fired, there was no errors, so figured I could leave it as is.